src/test/java/echoServer/MockClientWrapperTest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package echoserver;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import static org.junit.Assert.*;
+
+import java.io.*;
+import java.net.*;
+
+public class MockClientWrapperTest {
+    MockClientSocketWrapper mockClientSocketWrapper;
+
+  @Before public void initialize() {
+    String ip = "localhost";
+    int port = 4242;
+    BufferedReader input = new BufferedReader(new StringReader("hiii\n"));
+    PrintWriter output = new PrintWriter(new StringWriter(), true);
+    mockClientSocketWrapper = new MockClientSocketWrapper(input, output);
+    Client client = new Client(mockClientSocketWrapper);
+    client.start(ip, port);
+    mockClientSocketWrapper.close();
+  }
+
+  @Test public void connectsToServerSocket() {
```

Comment:
I believe in most projects, the @Test goes in a separate line above the method definition line